### PR TITLE
Verify sign-off email address against committer email field

### PR DESCRIPTION
**Pull request by robbat2**
Fixes: https://bugs.gentoo.org/show_bug.cgi?id=703684
Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

### CODE REVIEW

File: assign-pull-requests.py [hunk 1]
```diff
@@ -125,6 +125,33 @@ def main(ref_repo_path):
...
+
+    Other sign-off lines MAY also be present, but do not affect the outcome of
+    this check. It's entirely valid for one person to have multiple sign-off
+    lines, using different email addresses (signing off on something for both
+    work and Gentoo development)
+    """
+    lines = commit.message.splitlines()
+    committer_email = commit.committer.email.lower() # laziness here
+    # Get all lines that start with signed-off-by
+    signofflines = [x for x in lines if RE_SIGNOFFBY.match(x)]
```
Comment (mgorny) [inline assign-pull-requests.py]: Regex seems like a major overkill here.

File: assign-pull-requests.py [hunk 2]
```diff
@@ -125,6 +125,33 @@ def main(ref_repo_path):
...
+    for line in signofflines:
+        # Split the line, grab everything with '@' in it
+        possible_emails = [word for word in re.split(r'\s+', line) if '@' in word]
```
Comment (mgorny) [inline assign-pull-requests.py]: Do we really want to hack it like this? I'd say using [email.utils.parseaddr](https://docs.python.org/3/library/email.utils.html#email.utils.parseaddr) would be better.

**Review by mgorny — CHANGES_REQUESTED** [review]
I'm sorry for missing this earlier.

Also, please follow PEP 8 formatting.

File: assign-pull-requests.py [hunk 3]
```diff
@@ -126,6 +127,25 @@ def main(ref_repo_path):
     return 0
 
 
+def commit_contains_correct_signoff(commit):
+    """ Given a commit, it is considered to contain a correct sign-off IFF:
```
Comment (mgorny) [inline assign-pull-requests.py]: Let's try to at least roughly follow PEP 257.

File: assign-pull-requests.py [hunk 4]
```diff
@@ -126,6 +127,25 @@ def main(ref_repo_path):
...
+    AND that line contains the email address of the committer.
+
+    Other sign-off lines MAY also be present, but do not affect the outcome of
+    this check. It's entirely valid for one person to have multiple sign-off
+    lines, using different email addresses (signing off on something for both
+    work and Gentoo development)
+    """
+    committer_email = commit.committer.email.lower()  # case insensitive
+    for line in commit.message.splitlines():
+        if line.lower().startswith('signed-off-by:'):
```
Comment (mgorny) [inline assign-pull-requests.py]: I suppose you could store `line.lower()` in a helper var, and avoid the `.lower()` below too.

File: assign-pull-requests.py [hunk 5]
```diff
@@ -126,6 +127,25 @@ def main(ref_repo_path):
...
+            _signed_name, signed_email = email.utils.parseaddr(line)
```
Comment (mgorny) [inline assign-pull-requests.py]:
```suggestion
            _, signed_email = email.utils.parseaddr(line)
```

**Review by mgorny — APPROVED** [review]
LGTM (but note that I didn't test it).

**Comment by robbat2** [comment]
@mgorny  can you merge please?

**Comment by mgorny** [comment]
Done, thanks!

**Pull request closed without merging**